### PR TITLE
fix: data element operands [RWDQA-78]

### PR DESCRIPTION
**Pull request by tomzemp**
This PR adds handling for data element operands (see https://dhis2.atlassian.net/browse/RWDQA-78)

Here I'm assuming that all numerators will have a data element operand id (this seems to be required by the old app's UI). If that's not the case, the calculations could be modified slightly.

The configurations at https://demos.dhis2.org/dq/api/dataStore/dataQualityTool/settings have some dataElementOperandIDs where there is only an ID (i.e. not an `{deUID}.{cocUID}`), so I've also added handling in for retrieving the default COC and using that as the COC if the COC is missing from the data element operand id. I don't actually know why this occurs because the old app seems to sometimes saves a data element operand that is `{deUID}.{defaultCOCUID` and other times just saves `{deUID}`. This might be something to update based on how the numerator configuration gets implemented

In order to do these calculations, I've recalculated the calculations for Section 1C as this seemed easier to handle using some of the existing logic for extracting data values from responses.

### CODE REVIEW

File: src/utils/utils.js [hunk 1]
```diff
@@ -39,7 +39,8 @@ export const getConfigObjectsForAnalytics = (configurations, groupCode) => {
                 members.includes(numerator.code) &&
                 numerator.dataID &&
                 numerator.dataSetID &&
-                numerator.dataSetID.length
+                numerator.dataSetID.length &&
+                numerator.dataElementOperandID
```
Comment (tomzemp) [inline src/utils/utils.js:43]: requiring that all numerators will have a dataElementOperandID

File: src/components/annual-report/utils/utils.js [hunk 2]
```diff
@@ -38,8 +47,20 @@ export const convertAnalyticsResponseToObject = ({ headers, rows }) => {
             restructuredData[dx][ou] = {}
         }
 
-        if (!restructuredData[dx][ou][pe] && !isNaN(Number(rowData.value))) {
-            restructuredData[dx][ou][pe] = Number(rowData.value)
+        if (!restructuredData[dx][ou][pe]) {
+            if (co) {
+                restructuredData[dx][ou][pe] = {}
```
Comment (tomzemp) [inline src/components/annual-report/utils/utils.js:52]: this has been modified to allow for an additional level for coc in the data structure (but will behave as before if there is no `co` dimension in response)

File: src/components/annual-report/section1/useSectionOneData.js [hunk 3]
```diff
@@ -62,7 +73,7 @@ const reportQueries = {
         params: ({ dataElements, orgUnits, currentPeriod }) => ({
             dimension: `dx:${dataElements.join(';')},ou:${orgUnits.join(
                 ';'
-            )},pe:${currentPeriod}`,
+            )},pe:${currentPeriod},co`,
```
Comment (tomzemp) [inline src/components/annual-report/section1/useSectionOneData.js:76]: co is added as dimension as all data element operands will be data elements 😄 (passing `co` dimension with indicators will cause request to fail). If operand is missing coc, we will assume it's using default coc

File: src/components/annual-report/section1/section1Calculations.js [hunk 4]
```diff
@@ -717,13 +462,142 @@ const getSection1dChartInfo = ({ allOrgUnitsData, periodsIDs, ou }) => {
...
+    }, {})
+
+    const subOrgUnits =
+        expected_reports_by_org_unit_level.metaData?.dimensions?.ou ?? []
+
+    for (const de in mappedConfigurations.dataElementsAndIndicators) {
+        //  get overall values
+        const numerator = mappedConfigurations.dataElementsAndIndicators[de]
+        const threshold = numerator.missing
+        const [deOperand = de, coOperand = defaultCOC] =
```
Comment (KaiVandivier) [inline src/components/annual-report/section1/section1Calculations.js:520]: just a naming suggestion from what we used in the data entry app: `deID` and `cocID`

**Review by KaiVandivier — APPROVED** [review]
Looks good, and the results look correct 🙂 haven't tried it on an instance with actual categories other than `default` though

**Pull request merged**